### PR TITLE
feat(enumeration) bind to handle actions

### PR DESCRIPTION
**Pull request by talend-sromain**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

The enumeration component use only internal state to perform operation

**What is the new behavior?**

The eneumeration call handlers to make behaviours

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://319.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://319.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://319.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://319.talend.surge.sh)

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js [hunk 1]
```diff
@@ -285,14 +295,16 @@ class EnumerationWidget extends React.Component {
...
 
-		if (!this.valueAlreadyExist(value.value)) {
-			this.setState({
-				displayMode: 'DISPLAY_MODE_DEFAULT',
-				items: this.state.items.concat([{
-					values: [value.value],
-				}]),
-			}, this.setFormData.bind(this));
-			this.updateHeaderInputDisabled('');
+		if (!this.callActionHandler(ENUMERATION_ADD_ACTION, value.value)) {
```
Comment (jare-talend) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: Should not we trigger the event only if the value is not duplicated ?

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js [hunk 2]
```diff
@@ -304,7 +316,9 @@ class EnumerationWidget extends React.Component {
 	}
 
 	callActionHandler(actionName, value) {
-		if (this.props.registry.formContext.handleAction !== undefined) {
+		if (this.props.registry &&
```
Comment (jare-talend) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: Add one test on this

File: packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js (hunk 1 above)
Comment (talend-sromain) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: we can't know if the values are not all in the list & we want the same behaviour if it's in the list or not

Comment (jare-talend) [inline packages/forms/src/widgets/EnumerationWidget/EnumerationWidget.js]: Exact :)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://319.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://319.talend.surge.sh)

**Pull request merged**